api/api.go

```diff
@@ -221,16 +222,16 @@ func NewServerRoutes(s Server) *mux.Router {
 	m.Handle(lib.AEFeeds.String(), s.Middleware(remClientH.FeedsHandler))
 	m.Handle(lib.AEPreview.String(), s.Middleware(remClientH.DatasetPreviewHandler))
 
-	fsih := NewFSIHandlers(s.Instance, cfg.API.ReadOnly)
 	handleRefRouteMethods(m, lib.AEStatus, s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "fsi.status")), http.MethodGet, http.MethodPost)
-	m.Handle(lib.AEWhatChanged.String(), s.Middleware(fsih.WhatChangedHandler(lib.AEWhatChanged.NoTrailingSlash())))
-	m.Handle(lib.AEInit.String(), s.Middleware(fsih.InitHandler(lib.AEInit.NoTrailingSlash())))
-	m.Handle(lib.AECanInitDatasetWorkDir.String(), s.Middleware(fsih.CanInitDatasetWorkDirHandler(lib.AECanInitDatasetWorkDir.NoTrailingSlash())))
-	m.Handle(lib.AECheckout.String(), s.Middleware(fsih.CheckoutHandler(lib.AECheckout.NoTrailingSlash())))
-	m.Handle(lib.AERestore.String(), s.Middleware(fsih.RestoreHandler(lib.AERestore.NoTrailingSlash())))
-	m.Handle(lib.AEFSIWrite.String(), s.Middleware(fsih.WriteHandler(lib.AEFSIWrite.NoTrailingSlash())))
-	m.Handle(lib.AEFSICreateLink.String(), s.Middleware(fsih.CreateLinkHandler(lib.AEFSICreateLink.NoTrailingSlash())))
-	m.Handle(lib.AEFSIUnlink.String(), s.Middleware(fsih.UnlinkHandler(lib.AEFSIUnlink.NoTrailingSlash())))
+	handleRefRouteMethods(m, lib.AEWhatChanged, s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "fsi.whatchanged")), http.MethodGet, http.MethodPost)
+	m.Handle(lib.AEInit.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "fsi.init"))).Methods(http.MethodPost)
+	m.Handle(fmt.Sprintf("%s/{peername}/{name}", lib.AEInit), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "fsi.init"))).Methods(http.MethodPost)
```

Comment:
Tbh we abused that handleRef route a bit more than needed as we tack on that {selector} which is not really used in all that many endpoints. Yet we used it several times just to fill this usecase.
Think we should have this formatting extracted as a function and  have the inverse be the exception.
Happy to do that in one of my PRs, fine with it now (also getting that here specifically we don't want a full path passed on)